src/WebJobs.Script.WebHost/HostNameProvider.cs

```diff
@@ -21,13 +21,11 @@ namespace Microsoft.Azure.WebJobs.Script.WebHost
     public class HostNameProvider
     {
         private readonly IEnvironment _environment;
-        private readonly ILogger _logger;
         private string _hostName;
 
-        public HostNameProvider(IEnvironment environment, ILogger<HostNameProvider> logger)
```

Comment:
had to remove `ILogger` as a dependency here to avoid a cyclic dependency. `ILogger` has a dependency on `IEventGenerator` which if LinuxAppService, will have a dependency on `HostNameProvider`, and if that has a depdency on `ILogger`, it will end in an infinite recursion (no one wants that).

It makes sense that this provider shouldn't be logging things. And the caller can take care of it. To enforce logging, I added a parameter to the function.